api/src/operations/trigger/index.ts

```diff
@@ -17,9 +17,15 @@ export default defineOperationApi<Options>({
 		let result: unknown | unknown[];
 
 		if (Array.isArray(payloadObject)) {
-			result = await Promise.all(payloadObject.map((payload) => flowManager.runOperationFlow(flow, payload, context)));
+			result = await Promise.all(
+				payloadObject.map((payload) => {
+					const updatedData = { $trigger: payload, $last: payload };
+					return flowManager.runOperationFlow(flow, payload, { ...context, data: updatedData });
+				})
+			);
 		} else {
-			result = await flowManager.runOperationFlow(flow, payloadObject, context);
+			const updatedData = { $trigger: payload, $last: payload };
+			result = await flowManager.runOperationFlow(flow, payloadObject, { ...context, data: updatedData });
```

Comment:
You should be able to simply do something like

```js
flowManager.runOperationFlow(flow, payload, omit(context, 'data'));
```

here. That way, `FlowManager` will take care of setting `$trigger` and `$last` properly.

Reply:
Wow, that's super clean! Gotta remember that `omit` returns a new object instead of simply omitting. 😄 Thanks~~